conjure-java-core/src/integrationInput/java/com/palantir/product/EmptyPathServiceEndpoint.java

```diff
@@ -46,7 +47,12 @@ public void register(RoutingRegistry routingRegistry) {
         private class EmptyPathHandler implements HttpHandler {
             @Override
             public void handleRequest(HttpServerExchange exchange) throws IOException {
+                Long start = System.currentTimeMillis();
                 boolean result = delegate.emptyPath();
+                Long duration = System.currentTimeMillis() - start;
+                exchange.putAttachment(Metrics.DELEGATE_DURATION_KEY, duration);
+                exchange.putAttachment(Metrics.SERVICE_NAME_KEY, "EmptyPathService");
+                exchange.putAttachment(Metrics.RESOURCE_METHOD_NAME_KEY, "emptyPath");
```

Comment:
At the moment our generated <endpoint_name>Handler classes are only responsible for the bare minimum to read inputs, delegate to the service, and respond. Can we move these attachments to a separate handler like we do with the tracing handler?

Reply:
If we could provide service name and resource method to the RoutingRegistry we would be free to implement the handler in the runtime module

Reply:
Yes, we could do that and add that to the RoutingRegistry. Will adapt the PR.
I still think it would be valuable to get the duration of the delegate method in addition to the full request duration. For profiling of your resource code, you're probably interested in that value and not the rest. So I would be in favor to keep the duration attachment.

Reply:
I think we already get that data if the resource itself has been instrumented with Tritium, I'd prefer to focus on providing a replacement for the jersey metrics functionality here, and add additional features separately if that's alright with you.

Reply:
But tritium does it using reflection and proxying. It is also a pretty big dependency we could remove on the conjure part. Imho, It's a bit sad to not take advantage that using codegen we can avoid that, but we can always add it later.

Reply:
Sure, I don't think the service method invocations should necessarily be tied to undertow handler, we don't get leverage out of attaching invocation data to the HttpServerExchange. We could instead generate `InstrumentedFooService` as a concrete implementation which delegates to the service interface implementation. This way when services are called internally from code (as opposed to from top level http requests) we can still record metrics.

Reply:
That's an interesting idea:

Is this what you have in mind ?

```
//all generated from conjure service definition
interface FooService {
    Bar bar(...);
}

abstract class InstrumentedFooService implements FooService {
  abstract public Bar instrumentedBar(...);

  public final bar(...) {
    instrument(instrumentedBar(...));
  }
}
```

Reply:
That's an interesting idea:

Is this what you have in mind ?

```
//all below is generated from conjure service definition
interface FooService {
    Bar bar(...);
}

public final class InstrumentedFooService implements FooService {
  FooService fooService;
  ...

  of(FooService fooService) {...} 

  public final Bar bar(...) {
    instrument(fooService.bar(...));
  }
}
```

Reply:
Yep

Reply:
That makes a lot of sense. I guess we should do that in a separate PR once and if we want to remove the need for tritium.